### DISCUSSION TITLE
CMakeBuilder example

### DISCUSSION THREAD

**Discussion by mikesongming** [beeware/mobile-forge#31] [Q&A]
How to use CMakeBuilder to build wheels with binary extension with cmake tools?

**Comment by freakboy3742** [comment]
Right now, you don't. The mechanics are in place to recognise a CMake build by putting `cmake` into the `requirements.build` section of the `meta.yaml`; but that builder [is a stub](https://github.com/beeware/mobile-forge/blob/main/src/forge/build.py#L415). The implementation needs to be filled out so that the `_build()` method actually does something. There aren't any current recipes using CMake. 

Patches welcome :-)

**Comment by mikesongming** [comment]
I am looking for a solution to "forge" a binary extension managed by CMake in to iOS/MacOS. Current solutions all need to manually add cmake project into XCode project and configure building settings. With little experience on cross compiling iOS/MacOS using CMake, wondering if it is possible to be automated by Beeware and Mobile Forge.

**Reply by freakboy3742** [reply]
There shouldn't be any need to use xcode directly. CMake has native support for iOS; it just requires configuration to make it work. That configuration is something this forge is in a position to provide - as soon as someone works up with backend code.

**Reply by mikesongming** [reply]
You mean using toolchain like https://github.com/leetal/ios-cmake?

Could you explain how to "work up with backend code"? Maybe I can contribute some code :)

**Reply by freakboy3742** [reply]
I mean that CMake itself has [native support for iOS](https://cmake.org/features/), if it's configured correctly. The issue is that most projects won't be configuring it correctly; part of that problem is that the cross-platform build environment needs to present itself as an iOS target. That's what the [Mobileforge CMake builder](https://github.com/beeware/mobile-forge/blob/main/src/forge/build.py#L415) should be doing, but at present, it's a bare stub.

**Reply by mikesongming** [reply]
"The issue is that most projects won't be configuring it correctly", that is the daunting part. Maybe a predefined cmake template is needed? 

 [native support for iOS](https://cmake.org/features/), it is out of my knowledge -)
